WebDevStudios/nextjs-wordpress-starter
# Feature/#45 refactor dynamic validation

Part of  #45 

### Description

This feature refactors `getGfFieldValidationSchema` for scalability.

- Abstract `validationSchema` to `getValidationSchemaType(fieldData)` for determining which schema object type to create.
- Create `stringSchemaFactory` used for creating Yup string schema objects.
- Create get method `schema` for returning combined schema. Note: Schemas must be combined using `.concat` this way we can conditionally include various string methods from Yup.
- Setup method `getRequiredSchema` for `Yup.string().required`.
- Setup method `getMaxLengthSchema` for `Yup.string().max`.

### Screenshot
![Jan-08-2021 12-40-59](https://user-images.githubusercontent.com/25035900/104046924-cec35800-51ae-11eb-82d7-70a66066d501.gif)

### Verification
1. Checkout `test/#45-refactor-dynamic-validation` for testing.
2. Go to http://localhost:3000/blog/form-input-test and interact with the fields.
3. Check that code validation still works as expected.
